src/components/question-view/QuestionViewEditor.tsx

```diff
@@ -1,18 +1,61 @@
-import React, { useRef } from "react";
-import Editor from "@monaco-editor/react";
+import React, { useEffect, useRef, useState } from "react";
+import Editor, { Monaco } from "@monaco-editor/react";
+import { FeedbackType } from "./QuestionViewFeedback";
+import { getCodeFeedback } from "@/actions/getCodeFeedback";
 
 export default function QuestionViewEditor({
   handleEditorChange,
   language,
+  feedback,
+  shouldApplyDecorations,
 }: {
   handleEditorChange: (value: string | undefined) => void;
   language: string;
+  feedback: FeedbackType[];
+  shouldApplyDecorations: boolean;
 }) {
-  const editorRef = useRef(null);
+  const editorRef = useRef<any>(null);
+  const monacoRef = useRef<any>(null);
+  
+
+  function applyDecorations(editor: any, monaco: any, feedback: FeedbackType[]) {
```

Comment:
Not very good to leave it as type `any`. Found this stackoverflow https://stackoverflow.com/questions/70855979/react-typescript-types-for-monaco-editor which may help